jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/SsoSupport.kt

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.ide.BrowserUtil
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.ui.Messages
+import kotlinx.coroutines.withContext
+import software.aws.toolkits.core.credentials.sso.Authorization
+import software.aws.toolkits.core.credentials.sso.DiskCache
+import software.aws.toolkits.core.credentials.sso.SsoCache
+import software.aws.toolkits.core.credentials.sso.SsoLoginCallback
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+
+/**
+ * Shared disk cache for SSO for the IDE
+ */
+val diskCache by lazy { DiskCache() }
+
+object SsoPrompt : SsoLoginCallback {
```

Comment:
Thoughts on putting this in the explorer window instead? it's weird to say "connect" then have to move your mouse to the top

Reply:
Explorer may not always be available, such as new project wizard or a modal is open.